examples/silentpayments.c

```diff
@@ -0,0 +1,543 @@
+             *     2. Call `_silentpayments_recipient_scan_outputs`
+             *
+             */
+            secp256k1_silentpayments_found_output found_outputs[N_OUTPUTS];
+            secp256k1_silentpayments_found_output *found_output_ptrs[N_OUTPUTS];
+            secp256k1_silentpayments_recipient_public_data public_data;
+            secp256k1_pubkey spend_pubkey;
+            size_t n_found_outputs;
+            struct labels_cache labels_cache;
+
+            for (i = 0; i < N_OUTPUTS; i++) {
+                found_output_ptrs[i] = &found_outputs[i];
+            }
+            {
+                /** Labels setup
+                 *
+                 *  These steps are only necessary if the recipient is using the
+                 *  optional labels feature. Recipients not using labels can
+                 *  ignore these steps and simply pass `NULL` for the
+                 *  label_lookup and label_context arguments:
+                 *
+                 *      _silentpayments_recipient_scan_outputs(..., NULL, NULL);
+                 *
+                 *  In this case, since Bob has access to the full transaction
+                 *  outputs when scanning, it's easy for him to scan with labels,
+                 *  as demonstrated below. For efficient scanning, Bob keeps a
+                 *  cache of every label he has previously used and uses a
+                 *  callback to check if a potential label exists in his cache.
+                 *  Since the labels are created using an incremental integer
+                 *  `m`, if Bob ever forgets how many labels he has previously
+                 *  used, he can pregenerate a large number of labels e.g.,
+                 *  0..100_000.
+                 */
+                size_t len = 33;
+                secp256k1_pubkey label;
+                unsigned int m = 1;
+
+                /* Load Bob's spend public key */
+                ret = secp256k1_ec_pubkey_parse(ctx,
+                    &spend_pubkey,
+                    bob_spend_pubkey,
+                    33
+                );
+
+                /* Add an entry to the cache. This implies Bob has previously
+                 * called
+                 * `secp256k1_silentpayments_recipient_create_labelled_spend_pubkey`
+                 * and is using the resulting labelled spend pubkey to encode a
+                 * labelled silent payments address.
+                 */
+                ret &= secp256k1_silentpayments_recipient_create_label(ctx,
+                    &label,
+                    labels_cache.entries[0].label_tweak,
+                    bob_scan_key,
+                    m
+                );
+                ret &= secp256k1_ec_pubkey_serialize(ctx,
+                    labels_cache.entries[0].label,
+                    &len,
+                    &label,
+                    SECP256K1_EC_COMPRESSED
+                );
+                assert(ret);
```

Comment:
There should be an assert(ret) right after `recipient_create_label` because we're using its output as input to the next function.